### ISSUE TITLE
Usb(NotFound) when receiving

### ISSUE THREAD

**Issue by luizribeiro** [newAM/hackrfone#9]
Hi! Thanks for your work on this library.

I've been trying to get it to work with macOS and I haven't been able to receive anything. `rx()` always gives me an `Err(Usb(NotFound))`, while `device_version()` seems to work correctly.

This is the basic example I've tried running:

```rust
use hackrfone::{HackRfOne, RxMode, UnknownMode};

fn main() {
    let radio: HackRfOne<UnknownMode> = HackRfOne::new().unwrap();
    println!("{:?}", radio.device_version());

    let mut radio: HackRfOne<RxMode> = radio.into_rx_mode().unwrap();
    radio.rx().unwrap();
}
```

Which gives me this output:

```
Version(1, 0, 4)
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Usb(NotFound)', src/main.rs:8:16
```

I also tried your `rx.rs` example and got this:

```
Spawned sample thread
Sample thread disconnected
Shutting down sample thread
thread 'main' panicked at 'Failed to send exit event to sample thread: "SendError(..)"', src/main.rs:89:10
```

Have you encountered anything like this? Which platforms have you been able to test your library against?

**Comment by newAM** [comment]
I have onky used Linux to test this out. Unfortunately I do not have any Mac platforms available so I cannot test it out with that :/

If the original libhackrf works then this _should_ in theory also work the same because `rusb` is just a wrapper for libusb which is what libhackrf is using.

If I recall correctly the device version is reported in a header during USB negotiation, which may explain why the version works, but the RX endpoint does not.

As for the `SendError` that is probably just my crappy code 😅.  It works for me in the positive path, but I have done zero testing with what happens when the sample thread stops before the main thread tells it to, which is probably the issue there.

Some TODOs for me:
* [ ] Add tested platforms in README.
* [ ] Fix example for negative path.

**Comment by newAM** [comment]
Took a look at the example failure, that's my crummy code, sorry about that!

> A send operation can only fail if the receiving end of a channel is disconnected, implying that the data could never be received. The error contains the data being sent as a payload so it can be recovered.

If the sample thread ran up to here:
https://github.com/newAM/hackrfone/blob/4c7edbc2249a87d6c91a69456ab8e9f20bae05b4/examples/rx.rs#L36-L42

Then failed at the `rx()?` it would disconnect and later result in the `SendError` being unwrapped.

**Comment by newAM** [comment]
Hopefully this is fixed by #12 (included in v0.2.1), going to close this issue for now, feel free to re-open if it occurs again.
